src/common/CommonStructsAndConstants.h

```diff
@@ -0,0 +1,229 @@
+     */
+    u_int64_t numberOfEntries{0};
+
+    /**
+     * Stores the amount of lines in the indexed file. This value is also written after the indexing is done.
+     * However, the indexer and the extractor will work without it.
+     */
+    u_int64_t linesInIndexedFile{0};
+
+    /**
+     * Tell the IndexReader, if the entries are compressed. (1-byte padded to 8 bytes!) The value looks weird without
+     * the followup value when you look it up in written fqi files. C++ does not do a clean write in this case.
+     */
+    bool dictionariesAreCompressed{false};
+    Bytef placeholder[7]{0};
+
+    /**
+     * Reserved space for information which might be added in
+     * the future.
+     */
+    u_int64_t reserved[59]{0};
+
+    explicit IndexHeader(u_int32_t binaryVersion, u_int32_t sizeOfIndexEntry, u_int32_t blockInterval, bool dictionariesAreCompressed) {
+        this->indexWriterVersion = binaryVersion;
+        this->sizeOfIndexEntry = sizeOfIndexEntry;
+        this->blockInterval = blockInterval;
+        this->dictionariesAreCompressed = dictionariesAreCompressed;
+    }
+
+    IndexHeader() = default;
+
+    bool operator==(const IndexHeader &rhs) const;
+
+    bool operator!=(const IndexHeader &rhs) const {
+        return !(rhs == *this);
+    }
+
+    explicit operator bool() const;
+
+    bool operator!() { return !this; }
+};
+
+/**
+ * A (decompressed / processed) representation of an IndexEntryV[n].
+ */
+struct IndexEntry {
+    u_int64_t id{0};
+    u_int64_t offsetInRawFile{0};
+    u_int64_t startingLineInEntry{0};
+    u_int64_t compressedDictionarySize{0};
+    u_int32_t bits{0};
+    u_int16_t offsetOfFirstValidLine{0};
+    Bytef window[WINDOW_SIZE]{0};
+
+    IndexEntry(u_int64_t id,
+               u_int32_t bits,
+               u_int16_t offsetOfFirstValidLine,
+               u_int64_t offsetInRawFile,
+               u_int64_t startingLineInEntry);
+
+    explicit IndexEntry(u_int64_t id) {
+        this->id = id;
+    }
+
+    IndexEntry() = default;
+
+    bool operator==(const IndexEntry &rhs) const;
+
+    bool operator!=(const IndexEntry &rhs) const { return !(rhs == *this); }
+
+};
+
+struct VirtualIndexEntry {
+    bool operator==(const VirtualIndexEntry &rhs) const { return true; };
+};
+
+/**
+ * Represents an entry in a gz index file
+ * The order of the entries is set so, that we have a reduced padding of only 2 Bytes.
+ * Please take a look here for some explanations:
+ *   https://stackoverflow.com/questions/119123/why-isnt-sizeof-for-a-struct-equal-to-the-sum-of-sizeof-of-each-member
+ * The size of this struct is 8 Bytes.
+ */
+struct IndexEntryV1 : public VirtualIndexEntry {
+
+    /**
+     * The identifier of the raw compressed block for which this entry is.
+     */
+    u_int64_t blockID{0};
+
+    /**
+     * Block offset in raw gz file. Note, that this is not the absolute offset but a relative offset to the last entry.
```

Comment:
Maybe call it "relativeBlockOffsetInRawFile"? Actually a relative block offset is a block-start-distance-in-bytes or actually the size of the previous block in bytes, or not?

Reply:
Is the comment correct?